faucet/components/Faucet.tsx

```diff
@@ -0,0 +1,121 @@
+import { useState, useEffect } from 'react'
+import { isAddress } from 'web3-utils'
+import type { Networks } from '@hoprnet/hopr-ethereum/scripts/addresses'
+
+async function mint(
+  network: Networks,
+  address: string
+): Promise<{
+  success: boolean
+  transactionHash?: string
+  message?: string
+}> {
+  const response = await fetch(`/api/mint?network=${network}&address=${address}`)
+
+  if (response.status === 200) {
+    const { transactionHash } = await response.json()
+    return {
+      success: true,
+      transactionHash,
+    }
+  } else {
+    return {
+      success: false,
+      message: response.statusText,
+    }
+  }
+}
+
+function Faucet({ network }: { network: Networks }) {
+  const [address, setAddress] = useState<string>(undefined)
+  const [isValidAddress, setIsValidAddress] = useState<boolean>(undefined)
+  const [status, setStatus] = useState<'PENDING' | 'SUCCESS' | 'FAILURE'>(undefined)
+  const [message, setMessage] = useState<string>(undefined)
+  const isButtonDisabled = !isValidAddress || status === 'PENDING'
+  const showStatusMessage = status === 'SUCCESS' || status === 'FAILURE'
+
+  // update 'isValidAddress' on 'address' change
+  useEffect(() => {
+    if (typeof address === 'undefined' || address === '') {
+      setIsValidAddress(undefined)
+    } else {
+      setIsValidAddress(isAddress(address))
+    }
+  }, [address])
+
+  // once button is clicked
+  const onClick = async () => {
+    setStatus('PENDING')
+
+    const result = await mint(network, address)
+
+    if (result.success) {
+      setMessage(result.transactionHash)
+      setStatus('SUCCESS')
+    } else {
+      setMessage(result.message)
+      setStatus('FAILURE')
+    }
+  }
```

Comment:
First, lets make sure we always pass the variables we will be using in the function we call as to avoid leaks and global variables usage i.e. lets pass `network` and `address` to the function. Also, we should be adding a type to our functions.
```suggestion
  const onClick = async (network, address): void => {
    setStatus('PENDING')
    const { message, status } = await mint(network, address)
      .then(res => result.success ?  
        { message: result.transactionHash, status: 'SUCCESS' } : 
        { message: result.message, status: 'FAILURE' );
     setMessage(message);
     setStatus(status);
  }
```
This pattern can help us to start [currying](https://en.wikipedia.org/wiki/Currying) our functions to make behaviour more predictable.